packages/workflow/src/errors/abstract/execution-base.error.ts

```diff
@@ -0,0 +1,47 @@
+import type { IDataObject, JsonObject, Severity } from '../../Interfaces';
+import { ApplicationError } from '../application.error';
+
+interface ExecutionBaseErrorOptions {
+	cause?: Error | JsonObject;
+}
+
+export abstract class ExecutionBaseError extends ApplicationError {
+	description: string | null | undefined;
+
+	cause: Error | JsonObject | undefined;
```

Comment:
I think we should update this to be 2 separate properties, and stop conflating the 2 distinct pieces of information.
`cause` should not be used to store `JSONObject`. We don't have to make this change in this PR, but we should make this change soon, to make errors more type-safe, and also to be able to clearly distinguish between a `cause`, and remote json responses in the UI.